test/parallel/test-fs-rmdir-recursive.js

```diff
@@ -10,18 +10,56 @@ let count = 0;
 
 tmpdir.refresh();
 
-function makeNonEmptyDirectory() {
-  const dirname = `rmdir-recursive-${count}`;
-  fs.mkdirSync(path.join(dirname, 'foo', 'bar', 'baz'), { recursive: true });
+function makeNonEmptyDirectory(depth, files, folders, dirname, createSymLinks) {
```

Comment:
I really like this approach of building this out into a factory, so that it's easy to create a bunch of different scenarios, thanks for doing this :tada: